main.cc

```diff
@@ -1669,6 +1673,25 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 });
             };
 
+            auth::service_config maintenance_auth_config;
+            maintenance_auth_config.authorizer_java_name = sstring{auth::allow_all_authorizer_name};
+            maintenance_auth_config.authenticator_java_name = sstring{auth::allow_all_authenticator_name};
+            maintenance_auth_config.role_manager_java_name = sstring{auth::maintenance_socket_role_manager_name};
+
+            maintenance_auth_service.start(perm_cache_config, std::ref(qp), std::ref(mm_notifier), std::ref(mm), maintenance_auth_config, db::maintenance_socket_enabled::yes).get();
+
+            scheduling_group_key_config maintenance_cql_sg_stats_cfg =
+            make_scheduling_group_key_config<cql_transport::cql_sg_stats>(db::maintenance_socket_enabled::yes);
```

Comment:
Can someone fluent with the language of scheduling groups review this part?
@denesb?

Reply:
or @avikivity

Reply:
The second line in a multi-line statement needs a double indent.

But, why do we need separate statistics? The maintenance port will usually be quiet. We can just allow them to share statistics. Pushing out statistics comes with a large cost, and these are already split over service levels.

Reply:
Ok I understand this now after reading seastar test for scheduling group specific values. (The comments somehow didn't help me)

`scheduling_group_key_create` creates a map: the key is the scheduling group, the value is of type specified by `make_scheduling_group_key_config`.

So we have a value of given type in each scheduling group, and we can access the value of a given scheduling group using `scheduling_group_get_specific`.

So this will create a new stats object per scheduling group. This per-sg stats object will be then used by the new controller.

It's true that we create a new stats object, but these stats are write-only -- we don't register metrics based on them. The stats object is just there to avoid `if`fing every place we update stats in controller code (if stats object present / not used by maintenance socket, update stats...)